src/components/change-wallet/AddressRow.js

```diff
@@ -1,86 +1,187 @@
 import { colors, fonts } from '../../styles';
-import { abbreviations } from '../../utils';
+import { getFontSize } from '../../styles/fonts';
 import { ButtonPressAnimation } from '../animations';
+import { BottomRowText } from '../coin-row';
 import CoinCheckButton from '../coin-row/CoinCheckButton';
+import { ContactAvatar } from '../contacts';
 import { Column, Row } from '../layout';
 import { TruncatedAddress } from '../text';
 
 const sx = StyleSheet.create({
   accountLabel: {
     color: colors.dark,
     fontFamily: fonts.family.SFProText,
-    fontSize: Number(fonts.size.smedium.replace('px', '')),
+    fontSize: getFontSize(fonts.size.lmedium),
     fontWeight: fonts.weight.medium,
-  },
-  addressAbbreviation: {
-    fontFamily: fonts.family.SFProText,
-    opacity: 0.5,
-    textTransform: 'lowercase',
-    width: '100%',
+    paddingBottom: 5,
   },
   accountRow: {
-    marginLeft: 0,
+    flex: 1,
+    justifyContent: 'center',
+    marginLeft: 20,
   },
-  subItem: {
-    marginLeft: 15,
+  leftContent: {
+    flexDirection: 'row',
+    flex: 1,
   },
   rightContent: {
-    marginLeft: 60,
+    flex: 0,
+    marginRight: 19,
   },
   coinCheck: {
+    width: 30,
+    alignSelf: 'flex-end',
     marginTop: -15,
+    backgroundColor: 'red',
+    marginRight: 15,
+  },
+  borderBottom: {
+    marginLeft: 20,
+    paddingVertical: 10,
+    borderColor: colors.lightestGrey,
+    borderBottomWidth: StyleSheet.hairlineWidth,
+  },
+  readOnlyText: {
+    color: colors.mediumGrey,
+    fontWeight: fonts.weight.medium,
   },
-  isOnlyAddress: {
-    paddingTop: 15,
+  gradient: {
+    marginTop: 20,
+    paddingVertical: 3,
+    paddingHorizontal: 8,
+    textAlign: 'center',
+    borderRadius: 30,
+    height: 25,
   },
 });
 
-export default function AddressRow({ data, onPress, onEditAddress }) {
-  const { address, index, isOnlyAddress, isSelected, label, wallet_id } = data;
+const gradientColors = [colors.lightGrey, colors.blueGreyDark];
+const gradientProps = {
+  pointerEvents: 'none',
+  style: sx.gradient,
+};
 
-  const onLongPress = useCallback(() => {
-    onEditAddress(wallet_id, address);
-  }, [address, onEditAddress, wallet_id]);
+const linearGradientProps = {
+  ...gradientProps,
+  end: { x: 1, y: 1 },
+  start: { x: 0, y: 0 },
+  colors: [
+    colors.alpha(gradientColors[0], 0.1),
+    colors.alpha(gradientColors[1], 0.08),
+  ],
+};
 
-  return (
+const OptionsIcon = ({ onPress }) => (
+  <ButtonPressAnimation
+    onPress={onPress}
+    scaleTo={0.9}
+    style={{
+      width: 22,
+      height: 22,
+      alignSelf: 'flex-end',
+      justifyContent: 'center',
+    }}
+  >
     <View
-      style={[
-        sx.subItem,
-        sx.accountRow,
-        isOnlyAddress ? sx.isOnlyAddress : null,
-      ]}
+      style={{
+        borderRadius: 22,
+        width: 22,
+        height: 22,
+        alignItems: 'center',
+        justifyContent: 'center',
```

Comment:
should use `<Centered>` component instead of having to declare these flex styles